pkg/standalone-utils/contracts/ERC4626CowSwapFeeBurner.sol

```diff
@@ -52,19 +55,34 @@ contract ERC4626CowSwapFeeBurner is CowSwapFeeBurner {
         IERC20 feeToken,
         uint256 exactFeeTokenAmountIn,
         IERC20 targetToken,
-        uint256 minTargetTokenAmountOut,
+        uint256 encodedMinAmountsOut,
         address recipient,
         uint256 deadline
     ) external override onlyProtocolFeeSweeper nonReentrant {
         // In this case we first pull the wrapped token, unwrap, and then proceed to burn by creating an order for
         // the underlying token.
         feeToken.safeTransferFrom(msg.sender, address(this), exactFeeTokenAmountIn);
 
-        // Redeem and overwrite inputs with new asset and unwrapped amount.
+        uint256 minTargetTokenAmountOut = uint128(encodedMinAmountsOut >> 128);
+        uint256 minERC4626AmountOut = uint128(encodedMinAmountsOut & type(uint128).max);
+
         IERC4626 erc4626Token = IERC4626(address(feeToken));
+        // Redeem and overwrite inputs with new asset and unwrapped amount.
         feeToken = IERC20(erc4626Token.asset());
+
+        uint256 feeTokenBalanceBefore = feeToken.balanceOf(address(this));
+
         exactFeeTokenAmountIn = erc4626Token.redeem(exactFeeTokenAmountIn, address(this), address(this));
 
+        uint256 feeTokenBalanceAfter = feeToken.balanceOf(address(this));
+        uint256 redeemAmount = feeTokenBalanceAfter - feeTokenBalanceBefore;
+
+        if (exactFeeTokenAmountIn < minERC4626AmountOut) {
+            revert AmountOutBelowMin(feeToken, exactFeeTokenAmountIn, minERC4626AmountOut);
+        } else if (redeemAmount < minERC4626AmountOut) {
+            revert AmountOutBelowMin(feeToken, redeemAmount, minERC4626AmountOut);
+        }
```

Comment:
```suggestion
        if (redeemAmount < minERC4626AmountOut) {
            revert AmountOutBelowMin(feeToken, redeemAmount, minERC4626AmountOut);
        }
```
It seems like this top one is checked twice? It's a little hard to follow (could use more comments). The tests pass when this is removed, so either it's really redundant, or we're missing a test.